{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Build a complete table of metadata and experimental details\n",
    "\n",
    "As originally defined, using only with isolation source and collection data, there exist duplicate rows. \n",
    "\n",
    "These duplicates were intended to record multiple extractions from the same sample due to problems. (technical replicates)\n",
    "\n",
    "However, source and date are still the same and it is only when joined to the sequencing run table that these additional extractions take on meaning.\n",
    "\n",
    "Below we join the tables and then add a replicate_number column to disambiguate.\n",
    "\n",
    "Mock and negative control records complicate matters.\n",
    "\n",
    "A unique index (sample_name) is constructed sysmetically from isolation_source, collection_date and replicate_number.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There were 885 rows in sample table\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>*sample_name</th>\n",
       "      <th>sample_title</th>\n",
       "      <th>bioproject_accession</th>\n",
       "      <th>*organism</th>\n",
       "      <th>*collection_date</th>\n",
       "      <th>*env_biome</th>\n",
       "      <th>*env_feature</th>\n",
       "      <th>*env_material</th>\n",
       "      <th>*geo_loc_name</th>\n",
       "      <th>*host</th>\n",
       "      <th>...</th>\n",
       "      <th>perturbation</th>\n",
       "      <th>samp_collect_device</th>\n",
       "      <th>samp_mat_process</th>\n",
       "      <th>samp_size</th>\n",
       "      <th>samp_store_temp</th>\n",
       "      <th>samp_vol_we_dna_ext</th>\n",
       "      <th>description</th>\n",
       "      <th>PigPen</th>\n",
       "      <th>Cohort</th>\n",
       "      <th>Dysentery</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ja31/14286</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gut microflora metagenomics</td>\n",
       "      <td>2017-01-31</td>\n",
       "      <td>Host- gut</td>\n",
       "      <td>Faecal</td>\n",
       "      <td>Faecal</td>\n",
       "      <td>NSW, Australia</td>\n",
       "      <td>Porcine</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-80C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Ja31/29962</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gut microflora metagenomics</td>\n",
       "      <td>2017-01-31</td>\n",
       "      <td>Host- gut</td>\n",
       "      <td>Faecal</td>\n",
       "      <td>Faecal</td>\n",
       "      <td>NSW, Australia</td>\n",
       "      <td>Porcine</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-80C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Ja31/29707</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gut microflora metagenomics</td>\n",
       "      <td>2017-01-31</td>\n",
       "      <td>Host- gut</td>\n",
       "      <td>Faecal</td>\n",
       "      <td>Faecal</td>\n",
       "      <td>NSW, Australia</td>\n",
       "      <td>Porcine</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-80C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Ja31/14322</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gut microflora metagenomics</td>\n",
       "      <td>2017-01-31</td>\n",
       "      <td>Host- gut</td>\n",
       "      <td>Faecal</td>\n",
       "      <td>Faecal</td>\n",
       "      <td>NSW, Australia</td>\n",
       "      <td>Porcine</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-80C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Ja31/29913</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gut microflora metagenomics</td>\n",
       "      <td>2017-01-31</td>\n",
       "      <td>Host- gut</td>\n",
       "      <td>Faecal</td>\n",
       "      <td>Faecal</td>\n",
       "      <td>NSW, Australia</td>\n",
       "      <td>Porcine</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-80C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  *sample_name  sample_title  bioproject_accession  \\\n",
       "0   Ja31/14286           NaN                   NaN   \n",
       "1   Ja31/29962           NaN                   NaN   \n",
       "2   Ja31/29707           NaN                   NaN   \n",
       "3   Ja31/14322           NaN                   NaN   \n",
       "4   Ja31/29913           NaN                   NaN   \n",
       "\n",
       "                     *organism *collection_date *env_biome *env_feature  \\\n",
       "0  Gut microflora metagenomics       2017-01-31  Host- gut      Faecal    \n",
       "1  Gut microflora metagenomics       2017-01-31  Host- gut      Faecal    \n",
       "2  Gut microflora metagenomics       2017-01-31  Host- gut      Faecal    \n",
       "3  Gut microflora metagenomics       2017-01-31  Host- gut      Faecal    \n",
       "4  Gut microflora metagenomics       2017-01-31  Host- gut      Faecal    \n",
       "\n",
       "  *env_material   *geo_loc_name    *host    ...    perturbation  \\\n",
       "0       Faecal   NSW, Australia  Porcine    ...             NaN   \n",
       "1       Faecal   NSW, Australia  Porcine    ...             NaN   \n",
       "2       Faecal   NSW, Australia  Porcine    ...             NaN   \n",
       "3       Faecal   NSW, Australia  Porcine    ...             NaN   \n",
       "4       Faecal   NSW, Australia  Porcine    ...             NaN   \n",
       "\n",
       "   samp_collect_device samp_mat_process samp_size samp_store_temp  \\\n",
       "0                  NaN              NaN       NaN            -80C   \n",
       "1                  NaN              NaN       NaN            -80C   \n",
       "2                  NaN              NaN       NaN            -80C   \n",
       "3                  NaN              NaN       NaN            -80C   \n",
       "4                  NaN              NaN       NaN            -80C   \n",
       "\n",
       "  samp_vol_we_dna_ext description PigPen Cohort Dysentery  \n",
       "0                 NaN         NaN     1A    NaN       0.0  \n",
       "1                 NaN         NaN     1A    NaN       0.0  \n",
       "2                 NaN         NaN     1A    NaN       0.0  \n",
       "3                 NaN         NaN     1A    NaN       0.0  \n",
       "4                 NaN         NaN     1A    NaN       0.0  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read in the sample metadata table as it current exists.\n",
    "\n",
    "samples = pd.read_excel('../source_data/pigs_samples_IDs_for_NCBI.xlsx', sheet_name=0, skiprows=14, skipfooter=0, index_col=None, dtype={'*sample_name': str})\n",
    "print('There were {} rows in sample table'.format(len(samples)))\n",
    "samples.head()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There were 928 rows in the DNA pate table\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>elution plate number</th>\n",
       "      <th>well position</th>\n",
       "      <th>date of sample</th>\n",
       "      <th>pig ID</th>\n",
       "      <th>date and pig ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P1</td>\n",
       "      <td>A1</td>\n",
       "      <td>Fe21</td>\n",
       "      <td>14194</td>\n",
       "      <td>Fe21/14194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P1</td>\n",
       "      <td>A2</td>\n",
       "      <td>Fe28</td>\n",
       "      <td>14286</td>\n",
       "      <td>Fe28/14286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P1</td>\n",
       "      <td>A3</td>\n",
       "      <td>Fe21</td>\n",
       "      <td>29644</td>\n",
       "      <td>Fe21/29644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P1</td>\n",
       "      <td>A4</td>\n",
       "      <td>Fe14</td>\n",
       "      <td>29898</td>\n",
       "      <td>Fe14/29898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P1</td>\n",
       "      <td>A5</td>\n",
       "      <td>Fe21</td>\n",
       "      <td>29679</td>\n",
       "      <td>Fe21/29679</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  elution plate number well position date of sample pig ID date and pig ID\n",
       "0                   P1            A1           Fe21  14194      Fe21/14194\n",
       "1                   P1            A2           Fe28  14286      Fe28/14286\n",
       "2                   P1            A3           Fe21  29644      Fe21/29644\n",
       "3                   P1            A4           Fe14  29898      Fe14/29898\n",
       "4                   P1            A5           Fe21  29679      Fe21/29679"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.options.display.max_rows = 10\n",
    "seqruns = pd.read_excel('../source_data/DNA_plates.xlsx', sheet_name=0, index_col=None, dtype={'date and pig ID': str})\n",
    "seqruns.drop(seqruns.columns[[3,5]], axis=1, inplace=True)\n",
    "print('There were {} rows in the DNA pate table'.format(len(seqruns)))\n",
    "seqruns.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def demux(df):\n",
    "    # replace the current sample_name with that made from isolation_source and collection_date\n",
    "    df['alt_sample_name'] = df['isolation_source'].astype(str) + df['*collection_date'].map(lambda v: v.strftime('-%y%m%d') if not pd.isnull(v) else '') \n",
    "\n",
    "    # + \"-\" + ncbi.replicate_number.astype(str)\n",
    "\n",
    "    def f(n):\n",
    "        \"\"\"\n",
    "        Create a incrementing counter of replicates when encountering duplicated rows.\n",
    "        \"\"\"\n",
    "        if len(n) > 1:\n",
    "           print('{} was duplicated {} times'.format(n.name, len(n)))\n",
    "        return pd.Series(np.arange(1, len(n)+1), index=n.index)\n",
    "\n",
    "    # add a new column which records the number of replicates of a given isolation source and collection date\n",
    "    df['replicate'] = 0\n",
    "    df.replicate = df.groupby('alt_sample_name')['alt_sample_name'].apply(f)\n",
    "    df['alt_sample_name'] = df['isolation_source'].astype(str) + df['*collection_date'].map(lambda v: v.strftime('-%y%m%d') if not pd.isnull(v) else '') + \"-\" + df.replicate.map('{:02d}'.format)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "def demux2(df):\n",
    "    # replace the current sample_name with that made from isolation_source and collection_date\n",
    "    df['sn'] =  df['*collection_date'].map(lambda v: v.strftime('%b%d/') if not pd.isnull(v) else 'NA/') + df['isolation_source'].astype(str)\n",
    "\n",
    "    def f(n):\n",
    "        \"\"\"\n",
    "        Create a incrementing counter of replicates when encountering duplicated rows.\n",
    "        \"\"\"\n",
    "        if len(n) > 1:\n",
    "           print('{} was duplicated {} times'.format(n.name, len(n)))\n",
    "        return pd.Series(np.arange(1, len(n)+1), index=n.index)\n",
    "\n",
    "    # add a new column which records the number of replicates of a given isolation source and collection date\n",
    "    df['replicate'] = 0\n",
    "    df.replicate = df.groupby('sn')['sn'].apply(f)\n",
    "    df['sn'] = df['*collection_date'].map(lambda v: v.strftime('%b%d/') if not pd.isnull(v) else 'NA/') + df['isolation_source'].astype(str)  + \"-\" + df.replicate.map('{:02d}'.format)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There were 922 rows in the joined table\n",
      "There were 911 rows after removal of plate/well duplicates\n",
      "14160-170201 was duplicated 2 times\n",
      "14162-170201 was duplicated 2 times\n",
      "14172-170216 was duplicated 2 times\n",
      "14174-170131 was duplicated 2 times\n",
      "14190-170131 was duplicated 3 times\n",
      "14192-170131 was duplicated 2 times\n",
      "14193-170131 was duplicated 2 times\n",
      "14194-170207 was duplicated 2 times\n",
      "14263-170131 was duplicated 2 times\n",
      "14265-170131 was duplicated 2 times\n",
      "14274-170131 was duplicated 2 times\n",
      "14284-170131 was duplicated 2 times\n",
      "14298-170131 was duplicated 2 times\n",
      "14320-170131 was duplicated 2 times\n",
      "29645-170131 was duplicated 2 times\n",
      "29652-170131 was duplicated 2 times\n",
      "29653-170131 was duplicated 2 times\n",
      "29667-170131 was duplicated 2 times\n",
      "29668-170131 was duplicated 2 times\n",
      "29694-170131 was duplicated 2 times\n",
      "29718-170131 was duplicated 2 times\n",
      "29743-170131 was duplicated 2 times\n",
      "29753-170131 was duplicated 2 times\n",
      "29754-170131 was duplicated 2 times\n",
      "29781-170131 was duplicated 2 times\n",
      "29796-170131 was duplicated 2 times\n",
      "29797-170131 was duplicated 2 times\n",
      "29847-170131 was duplicated 2 times\n",
      "29865-170131 was duplicated 2 times\n",
      "29934-170131 was duplicated 2 times\n",
      "29951-170131 was duplicated 2 times\n",
      "ColiGuard-180124 was duplicated 8 times\n",
      "MockCommunity-170814 was duplicated 9 times\n",
      "NegativeControl was duplicated 19 times\n",
      "Protexin-180124 was duplicated 8 times\n"
     ]
    }
   ],
   "source": [
    "# Using the existing sample_name as an index, join samples and plates. \n",
    "# rebuild an integer index\n",
    "cmb = samples.set_index('*sample_name',).join(seqruns.set_index('date and pig ID'), how='inner').reset_index()\n",
    "# given the old sample name column a better name than 'index'\n",
    "cmb.rename(columns={'index': 'old_sample_name'}, inplace=True)\n",
    "print('There were {} rows in the joined table'.format(len(cmb)))\n",
    "\n",
    "# Only one record should exist for a given plate and well.\n",
    "cmb.drop_duplicates(['elution plate number', 'well position'], inplace=True)\n",
    "print('There were {} rows after removal of plate/well duplicates'.format(len(cmb)))\n",
    "\n",
    "demux(cmb)\n",
    "cmb.to_csv('test.csv')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>old_sample_name</th>\n",
       "      <th>sample_title</th>\n",
       "      <th>bioproject_accession</th>\n",
       "      <th>*organism</th>\n",
       "      <th>*collection_date</th>\n",
       "      <th>*env_biome</th>\n",
       "      <th>*env_feature</th>\n",
       "      <th>*env_material</th>\n",
       "      <th>*geo_loc_name</th>\n",
       "      <th>*host</th>\n",
       "      <th>...</th>\n",
       "      <th>description</th>\n",
       "      <th>PigPen</th>\n",
       "      <th>Cohort</th>\n",
       "      <th>Dysentery</th>\n",
       "      <th>elution plate number</th>\n",
       "      <th>well position</th>\n",
       "      <th>date of sample</th>\n",
       "      <th>pig ID</th>\n",
       "      <th>alt_sample_name</th>\n",
       "      <th>replicate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>18Ja24/ColiGuard_P03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DPI</td>\n",
       "      <td>2018-01-24</td>\n",
       "      <td>Lab</td>\n",
       "      <td>Probiotic ColiGuard</td>\n",
       "      <td>Probiotic ColiGuard</td>\n",
       "      <td>NSW, Australia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P3</td>\n",
       "      <td>H1</td>\n",
       "      <td>18Ja24</td>\n",
       "      <td>ColiGuard_P3</td>\n",
       "      <td>ColiGuard-180124-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>18Ja24/ColiGuard_P04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DPI</td>\n",
       "      <td>2018-01-24</td>\n",
       "      <td>Lab</td>\n",
       "      <td>Probiotic ColiGuard</td>\n",
       "      <td>Probiotic ColiGuard</td>\n",
       "      <td>NSW, Australia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P4</td>\n",
       "      <td>D3</td>\n",
       "      <td>18Ja24</td>\n",
       "      <td>ColiGuard_P4</td>\n",
       "      <td>ColiGuard-180124-02</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>920</th>\n",
       "      <td>NA/neg.control_P10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Laboratory - UTS</td>\n",
       "      <td>NaT</td>\n",
       "      <td>Lab</td>\n",
       "      <td>negative control</td>\n",
       "      <td>negative control</td>\n",
       "      <td>NSW, Australia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P10</td>\n",
       "      <td>D10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>neg.control_P10</td>\n",
       "      <td>NegativeControl-18</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>921</th>\n",
       "      <td>NA/neg.control_P10.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Laboratory - UTS</td>\n",
       "      <td>NaT</td>\n",
       "      <td>Lab</td>\n",
       "      <td>negative control</td>\n",
       "      <td>negative control</td>\n",
       "      <td>NSW, Australia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P10</td>\n",
       "      <td>G6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>neg.control_P10.2</td>\n",
       "      <td>NegativeControl-19</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>911 rows × 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          old_sample_name  sample_title  bioproject_accession  \\\n",
       "0    18Ja24/ColiGuard_P03           NaN                   NaN   \n",
       "1    18Ja24/ColiGuard_P04           NaN                   NaN   \n",
       "..                    ...           ...                   ...   \n",
       "920    NA/neg.control_P10           NaN                   NaN   \n",
       "921  NA/neg.control_P10.2           NaN                   NaN   \n",
       "\n",
       "            *organism *collection_date *env_biome         *env_feature  \\\n",
       "0                 DPI       2018-01-24        Lab  Probiotic ColiGuard   \n",
       "1                 DPI       2018-01-24        Lab  Probiotic ColiGuard   \n",
       "..                ...              ...        ...                  ...   \n",
       "920  Laboratory - UTS              NaT        Lab     negative control   \n",
       "921  Laboratory - UTS              NaT        Lab     negative control   \n",
       "\n",
       "           *env_material   *geo_loc_name *host    ...    description  PigPen  \\\n",
       "0    Probiotic ColiGuard  NSW, Australia   NaN    ...            NaN     NaN   \n",
       "1    Probiotic ColiGuard  NSW, Australia   NaN    ...            NaN     NaN   \n",
       "..                   ...             ...   ...    ...            ...     ...   \n",
       "920     negative control  NSW, Australia   NaN    ...            NaN     NaN   \n",
       "921     negative control  NSW, Australia   NaN    ...            NaN     NaN   \n",
       "\n",
       "    Cohort Dysentery elution plate number well position date of sample  \\\n",
       "0      NaN       NaN                   P3            H1         18Ja24   \n",
       "1      NaN       NaN                   P4            D3         18Ja24   \n",
       "..     ...       ...                  ...           ...            ...   \n",
       "920    NaN       NaN                  P10           D10            NaN   \n",
       "921    NaN       NaN                  P10            G6            NaN   \n",
       "\n",
       "                pig ID      alt_sample_name replicate  \n",
       "0         ColiGuard_P3  ColiGuard-180124-01         1  \n",
       "1         ColiGuard_P4  ColiGuard-180124-02         2  \n",
       "..                 ...                  ...       ...  \n",
       "920    neg.control_P10   NegativeControl-18        18  \n",
       "921  neg.control_P10.2   NegativeControl-19        19  \n",
       "\n",
       "[911 rows x 36 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.options.display.max_rows = 5\n",
    "cmb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There were 960 file pairs\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DNA_plate</th>\n",
       "      <th>DNA_well</th>\n",
       "      <th>r1_filename</th>\n",
       "      <th>r2_filename</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P1</td>\n",
       "      <td>A1</td>\n",
       "      <td>MON5838/MON5838_1/plate_1_A1_S1_R1_001.fastq.gz</td>\n",
       "      <td>MON5838/MON5838_1/plate_1_A1_S1_R2_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P1</td>\n",
       "      <td>A10</td>\n",
       "      <td>MON5838/MON5838_73/plate_1_A10_S73_R1_001.fast...</td>\n",
       "      <td>MON5838/MON5838_73/plate_1_A10_S73_R2_001.fast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P1</td>\n",
       "      <td>A11</td>\n",
       "      <td>MON5838/MON5838_81/plate_1_A11_S81_R1_001.fast...</td>\n",
       "      <td>MON5838/MON5838_81/plate_1_A11_S81_R2_001.fast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P1</td>\n",
       "      <td>A12</td>\n",
       "      <td>MON5838/MON5838_89/plate_1_A12_S89_R1_001.fast...</td>\n",
       "      <td>MON5838/MON5838_89/plate_1_A12_S89_R2_001.fast...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P1</td>\n",
       "      <td>A2</td>\n",
       "      <td>MON5838/MON5838_9/plate_1_A2_S9_R1_001.fastq.gz</td>\n",
       "      <td>MON5838/MON5838_9/plate_1_A2_S9_R2_001.fastq.gz</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  DNA_plate DNA_well                                        r1_filename  \\\n",
       "0        P1       A1    MON5838/MON5838_1/plate_1_A1_S1_R1_001.fastq.gz   \n",
       "1        P1      A10  MON5838/MON5838_73/plate_1_A10_S73_R1_001.fast...   \n",
       "2        P1      A11  MON5838/MON5838_81/plate_1_A11_S81_R1_001.fast...   \n",
       "3        P1      A12  MON5838/MON5838_89/plate_1_A12_S89_R1_001.fast...   \n",
       "4        P1       A2    MON5838/MON5838_9/plate_1_A2_S9_R1_001.fastq.gz   \n",
       "\n",
       "                                         r2_filename  \n",
       "0    MON5838/MON5838_1/plate_1_A1_S1_R2_001.fastq.gz  \n",
       "1  MON5838/MON5838_73/plate_1_A10_S73_R2_001.fast...  \n",
       "2  MON5838/MON5838_81/plate_1_A11_S81_R2_001.fast...  \n",
       "3  MON5838/MON5838_89/plate_1_A12_S89_R2_001.fast...  \n",
       "4    MON5838/MON5838_9/plate_1_A2_S9_R2_001.fastq.gz  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files = pd.read_csv('../fq_well_and_plate.txt', sep=' ', names=['DNA_plate','DNA_well','filename'], index_col=['DNA_plate', 'DNA_well'])\n",
    "# we will convert this to one row per read-pair, rather than one row for R1 and one for R2\n",
    "files = files.groupby(['DNA_plate', 'DNA_well']).agg(['first', 'last'])\n",
    "# convert the pandas group to a new dataframe\n",
    "files = pd.DataFrame(files.reset_index().values, columns=['DNA_plate', 'DNA_well', 'r1_filename', 'r2_filename'])\n",
    "print('There were {} file pairs'.format(len(files)))\n",
    "files.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "cmb.rename(columns={'elution plate number': 'DNA_plate', 'well position': 'DNA_well'}, inplace=True)\n",
    "out = cmb.set_index(['DNA_plate', 'DNA_well']).join(files.set_index(['DNA_plate', 'DNA_well']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "out.to_csv('out.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
